doctr/utils/multithreading.py

```diff
@@ -29,8 +30,8 @@ def multithread_exec(func: Callable[[Any], Any], seq: Iterable[Any], threads: Op
 
     threads = threads if isinstance(threads, int) else min(16, mp.cpu_count())
     # Single-thread
-    if threads < 2:
-        results = map(func, seq)
+    if threads < 2 or os.environ['DOCTR_CONCURRENCY_DISABLE'] == 'true':
+        results = list(map(func, seq))
```

Comment:
As mentioned by Felix, this cannot be changed in this PR. It's a complicated/dangerous change, and there is a PR for that.

After giving some thought to it, we'll go with the option where we return a generator whatever the input data structure. The user will be free to cast it to whatever they want. (to give you a dangerous/silent issue: if you have a tuple, pass it to this, and check that the output type is still a tuple and not none to perform a conditional operation, that will be a problem).

Another solution would be to dynamically cast it to the input type but I'm a bit afraid of the consequences if someone passes a sequence-like / generator-like custom data structures (it can be processed technically by the function, but casting it back will meddle with custom user code)